python/GafferUI/PlugValueWidget.py

```diff
@@ -399,80 +457,88 @@ def _popupMenuDefinition( self ) :
 
 	def __plugDirtied( self, plug ) :
 
-		if plug.isSame( self.__plug ) :
-
-			self._updateFromPlug()
+		if plug in self.__plugs :
```

Comment:
> Can we ever end up with plug being a different python object than the one in self.__plugs.

We can.

> I vaguely remember from last time I was in this world that isSame was essential for reliability here?

`isSame()` is a necessary workaround for when you would otherwise use `is`. But that doesn't apply here because sets use `__hash__`. This side of things has been taken care of for us by the RefCounted binding :

https://github.com/ImageEngine/cortex/blob/master/test/IECore/RefCountedTest.py#L66-L152

Reply:
Perfect, thanks, good to know. Appreciate the link.